features/net/network-socket/NetworkStack.h

```diff
@@ -160,11 +160,12 @@ class NetworkStack
      *  This call is non-blocking. If accept would block,
      *  NSAPI_ERROR_WOULD_BLOCK is returned immediately.
      *
-     *  @param handle   Destination for a handle to the newly created sockey
+     *  @param handle   Destination for a handle to the newly created socket
      *  @param server   Socket handle to server to accept from
+     *  @param address  Destination for the remote address or NULL
      *  @return         0 on success, negative error code on failure
      */
-    virtual int socket_accept(nsapi_socket_t *handle, nsapi_socket_t server) = 0;
+    virtual int socket_accept(nsapi_socket_t *handle, nsapi_socket_t server, SocketAddress *address=0) = 0;
```

Comment:
I just pushed a patch to a fork from your fork as a part of looking at the other implementations: https://github.com/geky/mbed/commit/d2400fc798c31e747e9ccec7288864bcfa40e776

Fee free to steal it, or I'll make a pr onto this branch once nanostack is updated. (not sure if there is a better way to handle this)